Add JSON schema validation for jobs and targets

pglass opened the pull request.

jmvrbanac commented on rift/api/common/resources.py:
```diff
@@ -45,12 +47,22 @@ def load_body(self, req, validator=None):
+        err_msg = "JSON validation failed"
+        if error.path:
+            err_msg += ": invalid value for key {0}".format(list(error.path))
+        if len(error.message) <= 100:
```
Why less than 100? Shouldn't this just be an else?

jmvrbanac commented on rift/api/common/resources.py:
```diff
@@ -45,12 +47,22 @@ def load_body(self, req, validator=None):
+        if error.path:
+            err_msg += ": invalid value for key {0}".format(list(error.path))
+        if len(error.message) <= 100:
+            err_msg += " - " + error.message
```
```' - {}'.format(error.message) ``` is preferred as it'll properly cast the contents to a string.

jmvrbanac commented:
Overall, it looks pretty good. Could you possibly add a test or two to cover the new code?

pglass commented:
I fixed the things you mentioned, and added some tests. Specter is pretty nice! I just tested the validators themselves though, and not the `on_post()` methods... I was wondering if you had thoughts on how to unittest the API itself, or if there was a way to do that already via falcon.

jmvrbanac commented:
Awesome thanks!  So there are a number of ways to test the on_xxxx() methods. You can use a wsgi testing wrapper such as webtest or similar. Or because Falcon is quite straight forward and works off the passed in object references, you can create stub objects for req and resp and pass those into the on_xxxx() methods then assert on the assigned data.

jmvrbanac commented:
I'll merge this for now, but it would be good to get the coverage up on those resources.
